$NetBSD: patch-ak,v 1.4 2002/10/07 19:01:25 wennmach Exp $

Make arla work on -current (post NetBSD 1.6).

Do not print a cryptic message after xfs_mod has been modloaded, say it
in clear text.

--- xfs/bsd/xfs_wrap-bsd.c.~1.33.~	Mon Sep 30 21:58:30 2002
+++ xfs/bsd/xfs_wrap-bsd.c	Mon Sep 30 22:22:24 2002
@@ -31,6 +31,8 @@
  * SUCH DAMAGE.
  */
 
+#include <arla-version.h>
+
 #include <xfs/xfs_locl.h>
 #include <xfs/xfs_message.h>
 #include <xfs/xfs_fs.h>
@@ -205,7 +207,9 @@
 #elif defined(__FreeBSD__)
 MOD_DEV(xfs,LM_DT_CHAR,-1,&xfs_dev);
 #else
-#if !defined(__APPLE__)
+#if __NetBSD_Version__ >= 106080000
+MOD_DEV("xfs_mod","xfs_mod", NULL, -1, &xfs_dev, -1)
+#elif !defined(__APPLE__)
 MOD_DEV("xfs_mod",LM_DT_CHAR,-1,&xfs_dev)
 #endif
 #endif
@@ -302,8 +306,14 @@
     }
     ret = lkmdispatch(lkmtp, cmd);
     if(cmd == LKM_E_LOAD) {
+#if __NetBSD_Version__ >= 106080000
+	xfs_dev_major = _module.lkm_cdevmaj;
+#else
 	xfs_dev_major = _module.lkm_offset;
-	printf ("xfs: cdev: %d, syscall: %d\n",
+#endif
+	printf ("xfs (%s): "
+		"using cdev: %d, syscall: %d\n",
+		arla_version,
 		xfs_dev_major, xfs_syscall_num);
     }
     return ret;

